pkg/podvolume/restorer.go

```diff
@@ -213,3 +220,27 @@ func newPodVolumeRestore(restore *velerov1api.Restore, pod *corev1api.Pod, backu
 	}
 	return pvr
 }
+
+func getVolumesRepositoryType(volumes map[string]VolumeBackupInfo) (string, error) {
+	if len(volumes) == 0 {
+		return "", errors.New("empty volume list")
+	}
+
+	// the podVolumeBackups list come from one backup. In one backup, it is impossible that volumes are
+	// backed up by different uploaders or to different repositories. Asserting this ensures one repo only,
+	// which will simplify the following logics
+	repositoryType := ""
+	for _, backupInfo := range volumes {
+		if backupInfo.RepositoryType == "" {
+			return "", errors.New("invalid repository type among volumes")
+		}
+
+		if repositoryType == "" {
+			repositoryType = backupInfo.RepositoryType
+		} else if repositoryType != backupInfo.RepositoryType {
+			return "", errors.New("multiple repository type in one backup")
```

Comment:
Similarly, it would be helpful to put more info in the error message?

Reply:
Done changed the message to `"multiple repository type in one backup, current type %s, differential one [type %s, snapshot ID %s, uploader %s]"`